src/core/utils/hooks/useTheme.ts

```diff
@@ -48,49 +60,67 @@ export const useTheme = (
       matches: isDark,
     }: MediaQueryList | MediaQueryListEvent) => {
       if (isDark) {
-        addDarkTheme(ownerDocument);
+        addDarkTheme({ ownerDocument, highContrast });
       } else {
-        addLightTheme(ownerDocument);
+        addLightTheme({ ownerDocument, highContrast });
       }
     };
 
     switch (theme) {
       case 'light':
         prefersDarkMediaQuery?.removeEventListener?.('change', addOSTheme);
-        addLightTheme(ownerDocument);
+        addLightTheme({ ownerDocument, highContrast });
         break;
       case 'dark':
         prefersDarkMediaQuery?.removeEventListener?.('change', addOSTheme);
-        addDarkTheme(ownerDocument);
+        addDarkTheme({ ownerDocument, highContrast });
         break;
       case 'os':
         if (prefersDarkMediaQuery != undefined) {
           addOSTheme(prefersDarkMediaQuery);
           prefersDarkMediaQuery.addEventListener?.('change', addOSTheme);
         } else {
-          addLightTheme(ownerDocument);
+          addLightTheme({ ownerDocument, highContrast });
         }
         break;
       default:
         if (
           ownerDocument.documentElement.className.indexOf('iui-theme') === -1
         ) {
-          addLightTheme(ownerDocument);
+          addLightTheme({ ownerDocument, highContrast });
         }
     }
 
     return () => {
       prefersDarkMediaQuery?.removeEventListener?.('change', addOSTheme);
     };
-  }, [ownerDocument, theme]);
+  }, [highContrast, ownerDocument, theme]);
 };
 
-const addLightTheme = (ownerDocument: Document) => {
-  ownerDocument.documentElement.classList.add('iui-theme-light');
-  ownerDocument.documentElement.classList.remove('iui-theme-dark');
+const addLightTheme = ({
+  ownerDocument,
+  highContrast,
+}: Required<ThemeOptions>) => {
+  const classList = ownerDocument.documentElement.classList;
+  const currentTheme = Array.from(classList).find((cls) =>
+    cls.startsWith('iui-theme'),
+  );
+  if (currentTheme) {
+    classList.remove(currentTheme);
+  }
+  classList.add(`iui-theme-light${highContrast ? '-hc' : ''}`);
 };
 
-const addDarkTheme = (ownerDocument: Document) => {
-  ownerDocument.documentElement.classList.add('iui-theme-dark');
-  ownerDocument.documentElement.classList.remove('iui-theme-light');
+const addDarkTheme = ({
+  ownerDocument,
+  highContrast,
+}: Required<ThemeOptions>) => {
+  const classList = ownerDocument.documentElement.classList;
```

Comment:
What part? I don't want to access this classList up in the hook because it's too specific (later might change to data attribute).

If you're talking about `currentTheme` logic, it's just one line.

Reply:
How many differences you can find between these 2
![image](https://user-images.githubusercontent.com/81580355/161266684-50a80934-f196-4196-b5b4-f4bb80a13530.png)

![image](https://user-images.githubusercontent.com/81580355/161266714-c08dc8ee-7799-4a91-875f-6813746f0506.png)

So I am just saying have a single function `applyTheme` or something and share same code.

Reply:
well ok, done